crates/bevy_ecs/src/query/filter.rs

```diff
@@ -443,21 +443,17 @@ macro_rules! impl_or_query_filter {
             fn update_component_access(state: &Self::State, access: &mut FilteredAccess<ComponentId>) {
                 let ($($filter,)*) = state;
 
-                let mut _new_access = access.clone();
-                let mut _not_first = false;
+                let mut _new_access = FilteredAccess::empty();
+
                 $(
-                    if _not_first {
-                        let mut intermediate = access.clone();
-                        $filter::update_component_access($filter, &mut intermediate);
-                        _new_access.append_or(&intermediate);
-                        _new_access.extend_access(&intermediate);
-                    } else {
-                        $filter::update_component_access($filter, &mut _new_access);
-                        _new_access.required = access.required.clone();
-                        _not_first = true;
-                    }
+                    let mut intermediate = access.clone();
+                    $filter::update_component_access($filter, &mut intermediate);
+                    _new_access.append_or(&intermediate);
+                    _new_access.extend_access(&intermediate);
```

Comment:
I think this line can be removed

Reply:
I was also wondering about that, but then I noticed that filter can still access the change ticks of components, and this needs to conflict with other queries in the system that mutably access those change ticks.

Reply:
I'm not sure I understand, but maybe you could add a test that fails if this line is missing?

Currently, if I remove this line, all tests are passing

Reply:
I've added a test that fails when this line is commented.

TLDR: `Query<(), Or<(Changed<A>,)>>` has to conflict with `Query<&mut A>` because the first read the change trackers and the second one writes them. However if `_new_access.extend_access(&intermediate);` is omitted then the first one is considered as not reading anything and that makes them compatible.